nicolalandro_ntsnet-cub200_ntsnet.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: ntsnet
+summary: a fine grane model for image classification.
+image: nts-net.png
+author: Moreno Carraffini and Nicola Landro
+tags: [vision]
+github-link: https://github.com/nicolalandro/ntsnet-cub200/archive/master.zip
+featured_image_1: no-image
+featured_image_2: no-image
+accelerator: "cuda-optional"
+---
+
+```python
+import torch
+model = torch.hub.load('nicolalandro/ntsnet-cub200', 'ntsnet', pretrained=True, **{'topN': 6, 'device':'cpu', 'num_classes': 200})
+```
+
+### How to use on bird
+
+```
+from torchvision import transforms
+import torch
+import urllib
+from PIL import Image
+
+transform_test = transforms.Compose([
+    transforms.Resize((600, 600), Image.BILINEAR),
+    transforms.CenterCrop((448, 448)),
+    # transforms.RandomHorizontalFlip(),  # only if train
+    transforms.ToTensor(),
+    transforms.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+])
+
+
+model = torch.hub.load('nicolalandro/ntsnet-cub200', 'ntsnet', pretrained=True, **{'topN': 6, 'device':'cpu', 'num_classes': 200})
+model.eval()
+
+url = 'https://raw.githubusercontent.com/nicolalandro/ntsnet-cub200/master/images/nts-net.png'
+img = Image.open(urllib.request.urlopen(url))
+scaled_img = transform_test(img)
+torch_images = scaled_img.unsqueeze(0)
+
+with torch.no_grad():
+    top_n_coordinates, concat_out, raw_logits, concat_logits, part_logits, top_n_index, top_n_prob = model(torch_images)
+    
+    _, predict = torch.max(concat_logits, 1)
+    pred_id = predict.item()
+    print('bird class:', pred_id)
+```
+
+### How to train
```

Comment:
this section is not really in line with the other Hub cards, and should just be removed.

The central idea of Hub is to get quick usage of trained models and get tangible outputs.
For referring to training mechanism, they will come to your github page and read more

Reply:
so is not import to train it from yourself? Is not an interesting case? I can see "pretrained=True" so you can also use "pretrained=False" and if you do not want to train it i do not know why exist this possibility. Other people do not specify it because train a resnet or something else is standard.

Reply:
it's not that it's not important to train it yourself.
It's not important to put training mechanisms in the Hub cards, which are more for using model artifacts.
If people want to train, they will invariably come to your github repository to do so